include/alpaka/mem/buf/Traits.hpp

```diff
@@ -124,20 +130,91 @@ namespace alpaka
 #    pragma clang diagnostic pop
 #endif
 
-    //! Allocates pinned/mapped memory on host, accessible by the given device.
+    //! If supported, allocates stream-ordered memory on the given queue and the associated device.
+    //! Otherwise, allocates regular memory on the device associated to the queue.
+    //! Please note that stream-ordered and regular memory have different semantics:
+    //! this function is provided for convenience in the cases where the difference is not relevant,
+    //! and the stream-ordered memory is only used as a performance optimisation.
     //!
     //! \tparam TElem The element type of the returned buffer.
     //! \tparam TIdx The linear index type of the buffer.
     //! \tparam TExtent The extent type of the buffer.
-    //! \tparam TDev The type of device the buffer is accessible from.
+    //! \tparam TQueue The type of queue used to order the buffer allocation.
+    //! \param queue The queue used to order the buffer allocation.
+    //! \param extent The extent of the buffer.
+    //! \return The newly allocated buffer.
+    template<typename TElem, typename TIdx, typename TExtent, typename TQueue>
+    ALPAKA_FN_HOST auto allocAsyncBufIfSupported(TQueue queue, TExtent const& extent = TExtent())
+    {
+        if constexpr(hasAsyncBufSupport<alpaka::Dev<TQueue>, Dim<TExtent>>)
+        {
+            return allocAsyncBuf<TElem, TIdx>(queue, extent);
+        }
+        else
+        {
+            return allocBuf<TElem, TIdx>(getDev(queue), extent);
+        }
+
+        ALPAKA_UNREACHABLE(allocBuf<TElem, TIdx>(getDev(queue), extent));
+    }
+
+    //! Allocates pinned/mapped host memory, accessible by all devices in the given platform.
+    //!
+    //! \tparam TPltf The platform from which the buffer is accessible.
+    //! \tparam TElem The element type of the returned buffer.
+    //! \tparam TIdx The linear index type of the buffer.
+    //! \tparam TExtent The extent type of the buffer.
     //! \param host The host device to allocate the buffer on.
-    //! \param dev The device to make the memory accessible from.
     //! \param extent The extent of the buffer.
     //! \return The newly allocated buffer.
-    template<typename TElem, typename TIdx, typename TExtent, typename TDev>
-    ALPAKA_FN_HOST auto allocMappedBuf(DevCpu const& host, TDev const& dev, TExtent const& extent = TExtent())
+    template<typename TPltf, typename TElem, typename TIdx, typename TExtent>
+    ALPAKA_FN_HOST auto allocMappedBuf(DevCpu const& host, TExtent const& extent = TExtent())
     {
-        return trait::BufAllocMapped<TElem, Dim<TExtent>, TIdx, TDev>::allocMappedBuf(host, dev, extent);
+        return trait::BufAllocMapped<TPltf, TElem, Dim<TExtent>, TIdx>::allocMappedBuf(host, extent);
+    }
+
+    //! Checks if the host can allocate a pinned/mapped host memory, accessible by all devices in the given platform.
+    //!
+    //! TDev is the type of device to allocate the buffer on.
```

Comment:
This is probably a copy-paste leftover from the earlier checker?

Reply:
Fixed.